### PR TITLE
Btat 8543

### PR DESCRIPTION
**Pull request by JoNB7777**

### CODE REVIEW

File: app/views/templates/ChangeSuccessView.scala.html [hunk 1]
```diff
@@ -23,34 +24,48 @@
...
 
   @if(user.isAgent) {
 
     @model.agentEmail match {
-      case Some(email) => {<p class="govuk-body">@messages("changeSuccess.helpOne.digital.agent", email)</p>}
-      case _ => {<p class="govuk-body">@messages("changeSuccess.helpOne.paper.agent")</p>}
+      case Some(email) => {<p class="govuk-body">@messages("changeSuccess.helpOne.digital")</p>}
+      case _ => {
+        <p class="govuk-body">@messages("changeSuccess.helpOne.paper.agent")</p>
+}
```
Comment (lewishay) [inline app/views/templates/ChangeSuccessView.scala.html:45]: Could you just pop all this onto one line to be consistent with the case above it

File: it/connectors/VatSubscriptionConnectorISpec.scala [hunk 2]
```diff
@@ -48,7 +49,8 @@ class VatSubscriptionConnectorISpec extends IntegrationBaseSpec {
     firstName = Some("Dave"),
     lastName = Some("Taylor"),
     organisationName = Some("D Taylor's Cars"),
-    tradingName = Some("DT Autos")
+    tradingName = Some("DT Autos"),
+    contactPreference = None
```
Comment (lewishay) [inline it/connectors/VatSubscriptionConnectorISpec.scala:53]: Could you change this so that `contactPreference` is present and update the relevant JSON in `VatSubscriptionStub` to also populate it. It doesn't change the outcome of this test but it's nice to have an additional check that every part of the JSON reads is working, and is consistent with the other fields in this val.

File: test/views/templates/ChangeSuccessViewSpec.scala [hunk 3]
```diff
@@ -43,110 +44,114 @@ class ChangeSuccessViewSpec extends ViewBaseSpec {
...
 
         lazy implicit val document: Document = Jsoup.parse(view.body)
 
-        "have the page title provided by the model" in {
-          elementText(Selectors.title) shouldBe s"$exampleTitle - Your client’s VAT details - GOV.UK"
+        "have the correct first paragraph" in {
+          elementText(Selectors.paragraphOne) shouldBe "We’ll send a letter to your principal place of business " +
+            "within 15 working days, telling you whether we can accept the trading name."
         }
+      }
```
Comment (lewishay) [inline test/views/templates/ChangeSuccessViewSpec.scala:93]: Could you add an additional test under here for a user whose contact preference is `None` (i.e. unable to retrieve it), and just check the paragraph text

File: test/views/templates/ChangeSuccessViewSpec.scala [hunk 4]
```diff
@@ -43,110 +44,114 @@ class ChangeSuccessViewSpec extends ViewBaseSpec {
...
+    "an agent is performing the action" when {
 
-        "have a finish button with the correct text" in {
-          elementText(Selectors.button) shouldBe "Finish"
-        }
+      "the agent has an email address registered" when {
 
-        "have a finish button which navigates to the Change of Circs overview page" in {
-          element(Selectors.button).select("a").attr("href") shouldBe mockConfig.manageVatSubscriptionServicePath
+        "the contact preference is digital" should {
```
Comment (lewishay) [inline test/views/templates/ChangeSuccessViewSpec.scala:100]: This clause is unnecessary as this is already implied by "the agent has an email address registered" above it

**Review by lewishay — CHANGES_REQUESTED** [review]
Could you also address the unused import warnings shown by SBT, at least in the files you have changed. Run a `clean compile` or `clean test:compile` if you don't see these.
